commands/social/games/slots.js

```diff
@@ -32,16 +32,22 @@ class Slots extends Social {
   
   async run(message, args, level) { // eslint-disable-line no-unused-vars
     if (message.settings.socialSystem !== "true") return message.response(undefined, "The social system has been disabled.");
-
-    const results = machine.play();
-    const winnings = this.help.cost * results.totalPoints;
-    message.buildEmbed()
-      .setColor(message.guild.me.roles.highest.color || 5198940)
-      .setAuthor("Misaki Slots")
-      .setDescription(`${results.visualize(false)}\n\n${results.winCount === 0 ? `${message.member.displayName} has lost!\nBetter luck next time!` : `Whoa... ${message.member.displayName} won!`}\n\n${results.winCount === 0 ? "" : `You have won ₲${winnings.toLocaleString()}`}`)
-      .setTimestamp()
-      .send();
-    if (results.winCount > 0) return message.member.givePoints(winnings);
+    if (!message.member.inventory.tokens) return message.response(undefined, "Ba...Baka You don't have any tokens to play Slots, buy some with the Store command");
+    
+    try {
+      await message.member.takeItem("tokens", 1);
+      const results = machine.play();
+      const winnings = this.help.cost * results.totalPoints;
+      message.buildEmbed()
+        .setColor(message.guild.me.roles.highest.color || 5198940)
+        .setAuthor("Misaki Slots")
+        .setDescription(`${results.visualize(false)}\n\n${results.winCount === 0 ? `${message.member.displayName} has lost!\nBetter luck next time!` : `Whoa... ${message.member.displayName} won!`}\n\n${results.winCount === 0 ? "" : `You have won ₲${winnings.toLocaleString()}`}`)
+        .setTimestamp()
+        .send();
+      if (results.winCount > 0) return message.member.givePoints(winnings);
+    } catch (e) {
```

Comment:
i removed all try/catch in there for a duplication reason please do not change that